### ISSUE TITLE
match wrong callback when set two same path

### ISSUE THREAD

**Issue by lwzcwh** [Yellow-Camper/libevhtp#129]
https://github.com/criticalstack/libevhtp/blob/0f52e868d88391d0f61d180c136404da3079efd9/evhtp.c#L906
when we set two path  using evhtp_set_cb  like below，it will math wrong cb：
path1:   /abc/0                     cb1
path2：/abc/012                  cb2

when we call the rest server using path /abc/012  ,it will call cb1

**Comment by NathanFrench** [comment]
It also searches in  an ordered manner; have you tried putting the longest match first?

**Comment by lwzcwh** [comment]
Yeah, I tried, It works. But it's unreasonable to ask the user to use it in this way.

**Comment by lwzcwh** [comment]
> It also searches in an ordered manner; have you tried putting the longest match first?
yeah，I  tried, it works.  but I'm a low-level component, and the real user is someone else, and I can't ask them to use it like this. I need to find a way to circumvent this problem, or our upper-level code needs to be changed.

**Comment by NathanFrench** [comment]
I see. Any auto-sort would only apply to static URI's, I cannot sort based on regex or globs for obvious reasons. Is that OK?

**Comment by NathanFrench** [comment]
Hello @lwzcwh,

I pushed up a test branch for you here: https://github.com/criticalstack/libevhtp/tree/issue129 (diff here: https://github.com/criticalstack/libevhtp/compare/issue129)

I am leaving this as an optional compile-time flag, you must run `cmake` with the following:

`cmake -DEVHTP_SORT_CALLBACKS=ON ..` 

..along with any other options you would like. 

Since I do not want to affect any other code, this is not enabled by default. If this looks OK to you, I will merge.
